app/src/main/java/com/lucas/medicaltools/MainActivity.kt

```diff
@@ -68,20 +65,12 @@ class MainActivity() : MviActivity<MainView, MainPresenter>(), MainView {
 
     private fun renderOnCreateHappenedState(medicalTools: List<MedicalTool>?) {
         renderRecyclerView(medicalTools)
-        dismissSpinner()
     }
 
-
     private fun renderFilterState(medicalTools: List<MedicalTool>?) {
         renderRecyclerView(medicalTools)
     }
 
-    private fun renderLoadingState() {
-        showSpinner()
-        onCreateHappenedSubject.onNext("Time")
```

Comment:
Pleae don't add this in, because you'll have loading state > Filter > Loading state > filter ......